design/doc.go

```diff
@@ -2,3 +2,60 @@
 // using goa framework. This generated API accepts HTTP GET/POST/PATCH/PUT/DELETE requests
 // from multiple clients.
 package design
+
+import "strings"
+
+type fieldDesc struct {
+	desc         string
+	mandOnCreate bool
+	mandOnUpdate bool
+}
+
+// desc returns a field description and by default the field is marked as
+// optional in creation and update of resource.
+func desc(str ...string) fieldDesc {
```

Comment:
Do we really need to enforce use of `desc(...)` function call everywhere? If the field is not mandatory either on create or update, I think using a regular string should be allowed. If you are fine with this approach, the variadic function can be avoided and expect just one string argument.

Reply:
No no, that is gone by now. I first thought of a greater good. But `mandatoryOnCreate(str string) string` is sufficient enough.